wordpress-mobile/WordPress-Android
# Add Create Site notification tracking

This PR adds tracking for three events of the Create Site notification:

- **Notification scheduled:** This is done by tracking the `create_site_notification_scheduled` event inside `CreateSiteNotificationScheduler` (92866bdb435ddd2d035f5c4d001c31183c5da426).
- **Notification shown:** This is done by tracking the `notification_shown` event with a type of `create_site` inside `CreateSiteNotificationHandler` (d199b68c9bc140db5ab0efb2bb0d83ce7d496396).
- **Notification tapped:** This is done by passing a notification type argument to the `PendingIntent` of the notification, which then triggers an existing logic inside the `MainActivity` to track a `notification_tapped` event with a type of `create_site` (d70e117a95fcfdd8d7aaedd324396596b22d7dcb).

Check #15059 for more context on the Create Site notification.

### To test

To test if the notification is shown, I recommend changing the following line from `DAYS` to `MINUTES`.

https://github.com/wordpress-mobile/WordPress-Android/blob/f1f1f7a0311d3c62a51ab85b198fad0f3ad5b598/WordPress/src/main/java/org/wordpress/android/workers/CreateSiteNotificationScheduler.kt#L23

1. Clear app data.
1. Login with an account that **doesn't have any sites**.
1. Notice the main screen.
1. On `logcat`, notice the following message:

```
🔵 Tracked: create_site_notification_scheduled
```

5. Close the app.
6. Wait for the notification to show up and tap on it.
7. On `logcat`, notice the following messages:

```
🔵 Tracked: notification_shown, Properties: {"notification_type":"create_site"}
🔵 Tracked: notification_tapped, Properties: {"notification_type":"create_site"}
```

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing unit tests.

3. What automated tests I added (or what prevented me from doing so)
Unit tests.
